pytential/symbolic/mappers.py

```diff
@@ -249,8 +333,52 @@ def map_common_subexpression(self, expr):
                 expr.prefix,
                 expr.scope)
 
+# }}}
+
+
+# {{{ KernelTransformationRemover
+
+class KernelTransformationRemover(IdentityMapper):
+    r"""A mapper that removes the transformations from the kernel of all
+    :class:`~pytential.symbolic.primitives.IntG`\ s in the expression.
+
+    This includes source and target derivatives and other such transformations.
+    Any unnecessary kernel arguments are also removed from
+    :class:`~pytential.symbolic.primitive.IntG.kernel_arguments`.
+    """
```

Comment:
Describe example use case in docstring?

Reply:
Added something in aab3e5b. Let me know what you think.

Reply:
LGTM!